src/test/model-viewer-base-spec.ts

```diff
@@ -208,6 +208,32 @@ suite('ModelViewerElementBase', () => {
           expect(dataUrlMatcher.test(element.toDataURL())).to.be.true;
         });
       });
+
+      suite('toBlob', () => {
+        test('produces a blob', async () => {
+          const blob = await element.toBlob();
+          expect(blob).to.not.be.null;
+        });
+
+        test('can convert blob to object URL', async () => {
+          const blob = await element.toBlob();
+          const objectUrl = URL.createObjectURL(blob);
+          const objectUrlMatcher = /^blob\:https?:\//;
+          expect(objectUrlMatcher.test(objectUrl)).to.be.true;
+        });
+
+        test('has size', async () => {
+          const blob = await element.toBlob();
+          expect(blob.size).to.be.greaterThan(0);
+        });
+
+        test('uses fallbacks on unsupported browsers', async () => {
+          // Emulate unsupported browser
```

Comment:
What's the best way to do this?

Reply:
Maybe you can do 

```ts
const canvas = element[$canvas]
canvas.toBlob = undefined;
```
?

Reply:
We wrote a small `spy` helper that would probably be appropriate here:

https://github.com/GoogleWebComponents/model-viewer/blob/0e6582b6d6e30ceeb51576435b605425d95bf803/src/test/helpers.ts#L160-L192


Here is an example of a test that uses it:
https://github.com/GoogleWebComponents/model-viewer/blob/0e6582b6d6e30ceeb51576435b605425d95bf803/src/test/features/ar-spec.ts#L45-L63

The basic idea is it lets you easily replace a property descriptor on an object, and returns a function that restores the original property descriptor when called. You could use it to temporarily replace `toBlob` on `HTMLCanvasElement`:

```typescript
const restoreCanvasToBlob = spy(HTMLCanvasElement.prototype, 'toBlob', { value: undefined });
// do your test...
restoreCanvasToBlob();
```

Reply:
Awesome! Used this ^